test/tests/nodes_tests.py

```diff
@@ -64,21 +68,59 @@ def __init__(self):
             }
         ]
 
-    @test(groups=['nodes.test', 'test-nodes'])
+    def check_node_count(self):
+        Nodes().api1_1_nodes_get()
+        nodes = loads(self.__client.last_response.data)
+        if len(nodes) == NODE_COUNT:
+            return True
+        return False
+
+    @test(groups=['nodes.discovery.test'])
+    def test_nodes_discovery(self):
+        """ Testing Graph.Discovery completion """
+        if self.check_node_count():
+            LOG.warning('Nodes already discovered!')
+            return
+        self.__discovery_duration = datetime.now()
+        LOG.info('Wait start time: {0}'.format(self.__discovery_duration))
+        self.__worker = Worker(queue=QUEUE_GRAPH_FINISH,callbacks=[self.handle_graph_finish])
+        self.__worker.start()
+
+    def handle_graph_finish(self,body,message):
+        routeId = message.delivery_info.get('routing_key').split('graph.finished.')[1]
+        assert_not_equal(routeId,None)
+        Workflows().api1_1_workflows_get()
+        workflows = loads(self.__client.last_response.data)
+        for w in workflows:
+            injectableName = w['definition'].get('injectableName') 
+            if injectableName == 'Graph.Discovery':
```

Comment:
Probably want to hook on "Graph.SKU.Discovery" since that's what gets kicked off automatically. "Graph.Discovery" is a sub workflow of that one.